src/gaps/QueryFilterParser.ts

```diff
@@ -247,6 +274,17 @@ function parseDataType(retrieve: ELMRetrieve): string {
   return retrieve.dataType.replace(/^(\{http:\/\/hl7.org\/fhir\})?/, '');
 }
 
+/**
+ * Pulls out the resource type of a result type
+ *
+ * @param expression The expression to pull out resource type from.
+ * @returns FHIR ResourceType name.
+ */
+function parseElementType(expression: ELMExpression): string {
+  const elementType = (expression.resultTypeSpecifier as ListTypeSpecifier).elementType as NamedTypeSpecifier;
```

Comment:
If this isn't a ListTypeSpecifier of NamedTypeSpecifier or a NamedTypeSpecifier then perhaps we should gracefully not return anything. In the case it isn't or doesn't exist this could error out.